NeoML/src/Dnn/Layers/TiedEmbeddingsLayer.cpp

```diff
@@ -152,9 +154,15 @@ const CDnnBlob* CTiedEmbeddingsLayer::getEmbeddingsTable() const
 {
 	NeoAssert( channelIndex >= 0 );
 
-	const CMultichannelLookupLayer* embeddingsLayer =
-		CheckCast<CMultichannelLookupLayer>( GetDnn()->GetLayer( embeddingsLayerName ) );
-	return embeddingsLayer->GetEmbeddings( channelIndex );
+	return getLookUpLayer()->GetEmbeddings( channelIndex );
+}
+
+CMultichannelLookupLayer* CTiedEmbeddingsLayer::getLookUpLayer() const
+{
+	CMultichannelLookupLayer* embeddingsLayer;
+	embeddingsLayer = CheckCast<CMultichannelLookupLayer>(
+		const_cast<CDnn*>(GetDnn())->GetLayer(embeddingPath).Ptr());
+	return embeddingsLayer;
```

Comment:
I think, it can be avoid using the `const_cast<CDnn*>`, If you return `const CMultichannelLookupLayer*`

Reply:
It shoudnt be const because it must change when called from LearnOnce